invenio_search/ext.py

```diff
@@ -201,13 +212,16 @@ def load_entry_point_group_mappings(self, entry_point_group_mappings):
         for ep in iter_entry_points(group=entry_point_group_mappings):
             self.register_mappings(ep.name, ep.module_name)
 
-    def load_entry_point_group_templates(self, entry_point_group_templates):
+    def load_entry_point_group_templates(self, property_list):
         """Load actions from an entry point group."""
         result = []
-        for ep in iter_entry_points(group=entry_point_group_templates):
-            with self.app.app_context():
-                for template_dir in ep.load()():
-                    result.append(self.register_templates(template_dir))
+        for ep in iter_entry_points(group=property_list):
+            if callable(ep.load()):
```

Comment:
Done, thanks. 
I'll leave this one open because I would like your opinion. I did it in this way to keep it backward compatible. I don't like so much that the `mappings` were provided as a list in the endpoints, whereas the `templates` where provided as a function that returned a list. To be consistent, it would be better if both returned a list. 

What do you think if we put  inside the `if callable(loaded_ep)` a deprecation warning, and on a future major release, drop the support to provide a function?  I have the feeling that the only internal module using the templates is the statistics, so, if we migrate that one, we'll be in a consistent situation.

Reply:
In general, we try for all the entrypoint-based features to support both providing "direct" values and functions. The latter is especially useful in case we rely on configuration to dynamically define templates (like in `invenio-stats`). I don't see a way where we could avoid that in `invenio-stats`, since each instance can define new types of statistics events.

---

In the future, I would also like to double-down on the function-based definitions by not just having them return a list of "paths" to collect mappings from, but return actual mapping bodies, so we could do things like:

```python
def records_search_mapping(app):
    default_query_fields = app.config.get("DEFAULT_QUERY_FIELDS", ["_all"])
    creator_mapping = {
        "type": "object",
        "properties": {
            "name": {"type": "text"},
            "affiliation": {"type": "text"},
            "orcid": {"type": "keyword"},
        },
    }
    return {
        "_index": "records-v1.0.0",
        "_alias": "records",
        "mappings": {
            "properties": {
                "title": ...,
                "authors": creator_mapping,
                "contributors": creator_mapping,
                ...,
            },
        },
        "settings": {"index.query.default_field": default_query_fields},
    }
```

Then in the entrypoint definitions in `setup.cfg`:

```ini
invenio_search.mappings =
    records = my_records_module.mappings:records_search_mapping
```

This would allow us to have more of a "code as config" approach to mappings, or even use the `opensearch-dsl-py`'s [class-based definitions for mappings](https://opensearch.org/docs/latest/clients/python-high-level/#indexing-a-document).